app/views/contexts/context.py

```diff
@@ -39,12 +43,28 @@ def __init__(
             self._response_metadata,
         )
 
+        self._path_finder = PathFinder(
+            self._schema,
+            self._answer_store,
+            self._list_store,
+            self._progress_store,
+            self._metadata,
+            self._response_metadata,
+        )
+
+        if self._location:
+            self._routing_path = self._path_finder.routing_path(
```

Comment:
We should set the routing path to `None` then create it in `CalculatedSummaryContext`, otherwise you are creating it for all context classes when only Cal summary needs it.

Actually have a look at `SectionSummaryContext` that passes in routing path and location as it needs it, then you also won't get type hint issues.

Reply:
Updated so that it is similar to Section Summary 👍